File: 02-人工智慧/02-使用套件/02-現代AI套件/A4-groq/hello/groqChat.js
import Groq from "npm:groq-sdk"

let question =  Deno.args.join(' ')

const groq = new Groq({
    apiKey: Deno.env.get("GROQ_API_KEY")
});

async function getGroqChatCompletion() {
    return groq.chat.completions.create({
        messages: [
            {
                role: "user",
                content: question // "Explain the importance of fast language models"
            }
        ],
        model: "llama3-8b-8192"
    });
}

async function main() {
    const chatCompletion = await getGroqChatCompletion();
    // Print the completion returned by the LLM.
    console.log(chatCompletion.choices[0]?.message?.content || "");
}

main()

File: 03-機器學習/02-使用套件/03-使用模型/05-groq/hello/groqChat.py
import os
import sys
from groq import Groq

question = " ".join(sys.argv[1:])
print("問題：", question)

client = Groq(
    api_key=os.environ.get("GROQ_API_KEY"),
)

chat_completion = client.chat.completions.create(
    messages=[
        {
            "role": "user",
            "content": question,
        }
    ],
    model="llama3-8b-8192",
)

print(chat_completion.choices[0].message.content)
